daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/XMLUtils.scala

```diff
@@ -70,6 +70,34 @@ object XMLUtils {
   val NegativeInfinityString = "-INF"
   val NaNString = "NaN"
 
+  /**
+   * Converts a string to a float, including handling our INF, -INF, and NaN notations.
+   * @param s
+   * @return
+   */
+  def strToFloat(s: String): Float = {
+    s match {
+      case PositiveInfinityString => Float.PositiveInfinity
+      case NegativeInfinityString => Float.NegativeInfinity
+      case NaNString => Float.NaN
+      case _ => s.toFloat
+    }
+  }
+
+  /**
+   * Converts a string to a double, including handling our INF, -INF, and NaN notations.
+   * @param s
+   * @return
+   */
+  def strToDouble(s: String): Double = {
+    s match {
+      case PositiveInfinityString => Double.PositiveInfinity
+      case NegativeInfinityString => Double.NegativeInfinity
+      case NaNString => Double.NaN
+      case _ => s.toDouble
+    }
+  }
+
```

Comment:
The `NodeInfo.Float.fromXMLString` and `NodeInfo.Double.fromXMLString` functions do something similar with a little extra error checking.

Reply:
Actually NodeInfo is in daffodil-runtime1, and this is in daffodil-lib, so to avoid code duplication I can convert NodeInfo.Double.fromString to call this. That's the right thing because the NodeInfo things construct JDouble boxed things anyway.